docs/articles/documentation/test-api/selecting-page-elements/selectors.md

```diff
@@ -138,45 +142,149 @@ You can initialize a selector with any of these objects.
 
-Functions and CSS selector stings that initialize a selector can return multiple DOM nodes.
-In this instance, you must filter these nodes to select a single node that will eventually be returned by the selector.
-Use the [text](selector-options.md#optionstext) and [index](selector-options.md#optionsindex) options for this.
+test('My test', async t => {
+    const osCount            = await Selector('.column.col-2 label').count;
+    const submitButtonExists = await Selector('#submit-button').exists;
+
+    expect(osCount).eql(3);
+    expect(submitButtonExists).is.true;
+});
+
+```
 
-The [text](selector-options.md#optionstext) option specifies text content of the node that should be returned
-or a regular expression that matches this content.
+Note that selector property getters are asynchronous.
 
-The [index](selector-options.md#optionsindex) option specifies the index of the node that should be returned.
+## Combined Selectors
 
-The following example demonstrates how to use the `text` and `index` options.
+Selector api provides methods that can be combined together, making flexible functional-style selector system.
+
+ 
+### Filter Multiple DOM Nodes
+
+If selector returns multiple DOM nodes you must filter them to select a single node that will eventually be returned by the selector.
+Selector provides methods to filter DOM nodes by its index and text: 
+
+Method | Type | Description 
+------ | ----- | -----
+`nth(index)` | Selector | Creates selector for `index` element in the matching set.
+`withText(text)` | Selector | Creates selector that filters matching set by given `text`.
+`withText(re)` | Selector | Creates selector that filters matching set by given regular expression `re`.
 
 ```js
 import { expect } from 'chai';
 import { Selector } from 'testcafe';
 
-fixture `My fixture`
-    .page `http://www.example.com/`;
 
-const removeButton = Selector(cl => document.getElementsByClassName(cl), { text: 'Remove' });
-const secondButton = Selector(cl => document.getElementsByClassName(cl), { index: 1 });
+fixture `Example page`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const secondCheckBox = Selector('input[type=checkbox]').nth(1);
+    const windowsLabel   = Selector('label').withText('Windows');
 
-test('A shadowed remove button is focused', async t => {
-    expect(await removeButton('shadowed').focused).to.be.true;
+    await t
+        .click(secondCheckBox)
+        .click(windowsLabel);
 });
+```
 
-test('The second disabled button is visible', async t => {
-    expect(await secondButton('disabled').visible).to.be.true;
+If all nodes are filtered out, the selector returns `null`.
+
+### Find Elements by DOM Hierarchy
+
+Selector provide methods to find elements by DOM hierarchy in jQuery-style.
+
+Property | Description
+------ | -----
+`find(cssSelector)` | Finds all descendants of all nodes in the matching set and filters them either by CSS selector.
```

Comment:
Finds all descendants of all nodes in the matching set and filters them by specified CSS selector.